CHANGELOG.md

```diff
@@ -62,6 +62,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### New features
 
+* `jj config edit --user` and `jj config set --user` will now pick a default
+  config location if no existing file is found, potentially creating parent directories.
+
```

Comment:
This should go under 0.8.0 :) (I assume it's here because we released 0.8.0 after you started working on this PR).